app/services/discovery_engine/query/filters.rb

```diff
@@ -34,6 +34,14 @@ def parse_param(key, value)
       filter_type, filter_field = key.match(FILTER_PARAM_KEY_REGEX)&.captures
       return nil unless filter_type && value.present?
 
+      # Finder Frontend should be more resilient to invalid filter values, but we can't rely on that
+      # for the time being. This ensures that occasionally observed garbage parameters such as:
+      #
+      #   filter_world_locations[\\\\]=all
+      #
+      # are ignored.
+      return nil unless Array(value).all? { _1.is_a?(String) }
```

Comment:
I'm having a bit of trouble reading this. Is this converting the value into an array and then checking if that all of the values in the array are a string?

Also, could we do `return unless` rather than `return nil`?

Reply:
`Array(value)` returns `value` if it's already an array, otherwise it returns an array containing `value` as its only element. It's handy when you have something that can either be an individual thing or an array of things (as is the case with a lot of these query parameters). 💡 

I'd rather explicitly return `nil` here (even though a plain return is semantically equivalent) because in the happy path this method _does_ return something later on and just having a plain return in a guard clause early on in the method might give a reader the impression that it doesn't.

You're right that it doesn't help this method being readable though, and what exactly makes a parameter "valid" is secondary to this method so I've refactored it out into a helper function!

Reply:
Thanks for explaining. Putting it in a well named method has helped.